services/QuillLMS/engines/evidence/app/services/evidence/vertex_ai/text_classifier.rb

```diff
@@ -65,6 +73,16 @@ def run
           .max_by { |score, i| [score, i] }
           .last
       end
+
+      private def retry_with_exceptions(max_num_retries, exception_classes)
```

Comment:
Note for future, this is generic, so if we reuse this pattern in the future, it could be its own class/service for reusability..

Reply:
I thought about extracting it but I opted to wait until we get more examples of it.

Reply:
Yeah, I had the exact same thought. I originally wrote a "we should extract this!" comment, then thought, "meh, probably early, just flag it for later".